docs/book/src/development/conventions.md

```diff
@@ -36,3 +36,25 @@ use
 ```
 
 And then within the code you can check the length or range over the slice.
+
+## Tests
+
+There are two types of tests written for CAPA controllers in this repo:
+* Unit tests  
+* Integration tests
+* E2E tests
+
+If any new unit or integration tests has to be added in this repo,we should follow the below conventions.
+
+### Unit tests
+These tests are meant to verify the functions inside the same controller file where we perform sanity checks, functionality checks etc. Here we use all types of mocking like [fakeclient](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/client/fake) to mock the services API calls. 
```

Comment:
Both fakeclient and testEnv are used in unit tests, for example, in https://github.com/kubernetes-sigs/cluster-api-provider-aws/blob/main/controllers/awscluster_controller_unit_test.go. So we cannot say use fakeclient for unit test and testEnv for integration test. The choice depends on test cases as discussed in previous PRs.

Reply:
Thanks @pydctw for comments, but what do you suggest, shall we not mention about the fakeclient and envtest here?

Reply:
@pydctw wdyt now?

Reply:
Thankyou, if you dont mind can you lgtm it?